### DISCUSSION TITLE
Transaction with API calls

### DISCUSSION THREAD

**Discussion by mohoksoft** [btcpayserver/btcpayserver#4004] [Technical Support] [answered]
Does anyone know a list of API calls (Greenfield API ) to create and execute a transaction (shortest path)?

**Comment by bolatovumar** [comment]
@mohoksoft see here for how this can be done: https://docs.btcpayserver.org/API/Greenfield/v1/#operation/StoreOnChainWallets_CreateOnChainTransaction

**Reply by mohoksoft** [reply]
I've already tried the exact API call you're pointing to. I have listed the following in the body:
{
  "destinations": [
    {
      "destination": "bc1qmqlzk2sxnj6vfw72klwkweyaw9e6csqwjz6yc0",
      "amount": "0.00001",
      "subtractFromAmount": true
    }
  ],
  "feerate": 1,
  "proceedWithPayjoin": true,
  "proceedWithBroadcast": true,
  "noChange": false,
  "rbf": true,
  "excludeUnconfirmed": false,
  "selectedInputs": [
    "26e8eae2fcb761e978761a94b07f3e1b0a24e1200eed1e31ba2b1d40b3db86db00000000",
    "a84d62c104bd3274c0410c50039ad1f5a3742f1926d579e6ba8685864d8851bb00000000",
    "71202c95e42d7c3c4e74457904aabc4c4283e08bfb43a152a1cb3c217c20bf9400000000"
  ]
}

but response is:
{
    "code": "not-available",
    "message": "BTC sending services are not currently available"
}
Any help would be appreciated. Thanks.

**Reply by bolatovumar** [reply]
@mohoksoft it's because it's a read-only wallet. You can't send from a read-only wallet. You can tell if it's a read-only wallet by going to the wallet page on your BTCPay Server instance. If it doesn't have the "Send" button then it's a read-only wallet.

![Capture](https://user-images.githubusercontent.com/1934678/185036441-b625aeca-4f8f-46d8-a4e7-97dcccda8cfa.PNG)

**Reply by mohoksoft** [reply]
Unfortunately, it is not a read only wallet, because it has a Send button. I am also an admin on the server and I created the wallet. But thank you for your reply.

**Reply by Kukks** [reply]
The send button also appears for non-hot wallets since you can always sign with a seed manually.  Click on the settings icon and see if you can see "Hot wallet" or "View seed"
<img width="236" alt="image" src="https://user-images.githubusercontent.com/1818366/185051062-dc5d0f12-d03a-416f-9903-ec0dc9cb1ff8.png">

**Reply by mohoksoft** [reply]
When I click on Actions I don't see View Seed.
![image](https://user-images.githubusercontent.com/104525102/185140324-d122a1d7-a5c3-46e8-a888-3a7906354832.png)
What can I do to chance wallet settings?

**Reply by mohoksoft** [reply]
I was able to make a transaction by creating a Hot Wallet. Now API call work. You were right. Thanks guys.

**Comment by Kukks** [comment, marked as answer]
https://docs.btcpayserver.org/API/Greenfield/v1/#operation/StoreOnChainWallets_CreateOnChainTransaction
